### PR TITLE
loosen dependency versions

### PR DESCRIPTION
**Pull request by ddl-joyce-zhao**
## Description

Loosen several dependency libraries' versions.

## Related Issue
https://dominodatalab.atlassian.net/browse/DOM-55190

## Type of Change

- [ ] 📚 Examples / docs / tutorials / dependencies update
- [ ] 🔧 Bug fix (non-breaking change which fixes an issue)
- [ ] 🥂 Improvement (non-breaking change which improves an existing feature)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔐 Security fix

## Checklist

- [ ] I've read the [`CONTRIBUTING.md`](https://github.com/dominodatalab/domino-data/blob/main/CONTRIBUTING.md) guide.
- [ ] I've updated the code style using `make codestyle`.
- [ ] I've written tests for all new methods and classes that I created.
- [ ] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -42,20 +42,20 @@ classifiers = [  #! Update me
...
-backoff = "^1.11.1"
+backoff = ">=1.11.1"
 bson = "^0.5.10"
 urllib3 = "^1.26.16"
 
 [tool.poetry.group.dev.dependencies]
 Sphinx = "^5.3.0"
 bandit = "^1.7.5"
-black = {version = "^22.12.0", allow-prereleases = true}
+black = {version = "^24.3.0", allow-prereleases = true}
```
Comment (ddl-joyce-zhao) [inline pyproject.toml:58]: Upgraded to remove the vulnerability CVE-2024-21503

File: pyproject.toml [hunk 2]
```diff
@@ -42,20 +42,20 @@ classifiers = [  #! Update me
...
-pandas = "^1.3.0"
-httpx = "^0.23.0"
+pandas = ">=1.3.0"
+httpx = ">=0.23.0"
 attrs = ">=20.1.0"
 python-dateutil = "^2.8.0"
 pyarrow = "^14.0.2"
 loguru = "^0.5.3"
-backoff = "^1.11.1"
+backoff = ">=1.11.1"
```
Comment (ddl-joyce-zhao) [inline pyproject.toml:51]: upgraded to handle the conflicts with cohere==4.0 https://dominodatalab.atlassian.net/browse/DOM-56097

File: pyproject.toml [hunk 3]
```diff
@@ -42,20 +42,20 @@ classifiers = [  #! Update me
 
 [tool.poetry.dependencies]
 python = "^3.8"
-pandas = "^1.3.0"
-httpx = "^0.23.0"
+pandas = ">=1.3.0"
```
Comment (ddl-joyce-zhao) [inline pyproject.toml:45]: upgraded as per the users' request

File: pyproject.toml [hunk 4]
```diff
@@ -42,20 +42,20 @@ classifiers = [  #! Update me
...
+httpx = ">=0.23.0"
```
Comment (ddl-joyce-zhao) [inline pyproject.toml:46]: upgraded to handle the conflicts with cohere==4.0 https://dominodatalab.atlassian.net/browse/DOM-56097

File: datasource_api_client/__init__.py [hunk 5]
```diff
@@ -1,2 +1,3 @@
 """ A client library for accessing Datasource API """
+
```
Comment (ddl-joyce-zhao) [inline datasource_api_client/__init__.py:2]: The files are reformatted after black is upgraded.

**Review by ddl-gabrielhaim — APPROVED** [review]

**Pull request merged**